src/test/java/org/folio/rest/impl/TenantRefAPITest.java

```diff
@@ -0,0 +1,146 @@
+package org.folio.rest.impl;
+
+import static io.vertx.core.Future.succeededFuture;
+import static org.folio.rest.RestVerticle.OKAPI_HEADER_TENANT;
+import static org.folio.rest.RestVerticle.OKAPI_HEADER_TOKEN;
+
+import java.util.Collections;
+
+import javax.ws.rs.core.MediaType;
+
+import org.folio.rest.RestVerticle;
+import org.folio.rest.client.TenantClient;
+import org.folio.rest.jaxrs.model.LostItemFeePolicies;
+import org.folio.rest.jaxrs.model.LostItemFeePolicy;
+import org.folio.rest.jaxrs.model.OverdueFinePolicies;
+import org.folio.rest.jaxrs.model.OverdueFinePolicy;
+import org.folio.rest.jaxrs.model.Parameter;
+import org.folio.rest.jaxrs.model.TenantAttributes;
+import org.folio.rest.persist.PostgresClient;
+import org.folio.rest.tools.utils.NetworkUtils;
+import org.folio.rest.utils.OkapiClient;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import io.restassured.RestAssured;
+import io.restassured.http.Header;
+import io.restassured.specification.RequestSpecification;
+import io.vertx.core.DeploymentOptions;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.unit.Async;
+import io.vertx.ext.unit.TestContext;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+
+@RunWith(VertxUnitRunner.class)
+public class TenantRefAPITest {
+  private static final int PORT = NetworkUtils.nextFreePort();
+  private static Vertx vertx;
+
+  private final OkapiClient okapiClient = new OkapiClient(PORT);
+
+  @BeforeClass
+  public static void setUpClass(final TestContext context) throws Exception {
+    Async async = context.async();
+    vertx = Vertx.vertx();
+
+    PostgresClient.getInstance(vertx).startEmbeddedPostgres();
+    final TenantClient tenantClient = createTenantClient();
+
+    DeploymentOptions restDeploymentOptions = new DeploymentOptions()
+      .setConfig(new JsonObject().put("http.port", PORT));
+
+    vertx.deployVerticle(RestVerticle.class.getName(), restDeploymentOptions,
+      res -> {
+        try {
+          tenantClient.postTenant(getTenantAttributes(), result -> async.complete());
```

Comment:
`result -> async.complete()` should be replaced by `context.asyncAssertSuccess(result -> async.complete())` to fail the test when postTenant fails.